### PR TITLE
instrumentation of  @aws-sdk/client-dynamodb

### LINKED ISSUE #2958: AWS instrumentation for SDK v3, DynamoDB
Provide instrumentation for AWS SDK version 3 DynamoDB client

### PR DESCRIPTION
**Pull request by david-luna**
PR to add instrumentation to `@aws-sdk/client-dynamodb` package. Since in SDK v3 all clients are similar I took a same approach done in S3 intrumentation

- a soft guard is added for different spans triggering the middleware
- a guard is added is case `finalizeRequest` middleware is not executed

Closes #2958 

### Checklist

- [x] Implement code
- [x] Add tests
- [ ] Update TypeScript typings
- [ ] Update documentation
- [ ] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/main/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

File: test/_utils.js [hunk 1]
```diff
@@ -140,6 +140,28 @@ function sortApmEvents (events) {
     })
 }
 
+/**
+ * Slurp everything from the given ReadableStream and return the content,
+ * converted to a string.
+ */
+async function slurpStream (stream) {
```
Comment (david-luna) [inline test/_utils.js:147]: Note for reviewer: function copied here to be used in `use-client-*` files. Current `dynamodb` tests don't make use of it, yet. Update in `use-client-s3.js` pending

**Comment by david-luna** [comment]
Ci failing due to a problem in environment. Apparently we're using a newer version of `npm` with Nodejs <=14

```
ERROR: npm v9.8.0 is known not to run on Node.js v12.0.0.  This version of npm supports the following node versions: `^14.17.0 || ^16.13.0 || >=18.0.0`. You can find the latest version at [https://nodejs.org/.](https://nodejs.org/)
```

**Review by david-luna — COMMENTED** [review]
/test tav @aws-sdk/client-dynamodb

**Comment by david-luna** [comment]
> Ci failing due to a problem in environment. Apparently we're using a newer version of `npm` with Nodejs <=14
> 
> ```
> ERROR: npm v9.8.0 is known not to run on Node.js v12.0.0.  This version of npm supports the following node versions: `^14.17.0 || ^16.13.0 || >=18.0.0`. You can find the latest version at [https://nodejs.org/.](https://nodejs.org/)
> ```
> 
> This does not make much sense with the previous message shown by the GH action `setup-node@v3` which shows
> 
> ```
> Run actions/setup-node@v3
>   with:
>     node-version: 14.0
>     always-auth: false
>     check-latest: false
>     token: ***
> 
> Attempting to download 14.0...
> Acquiring 14.0.0 - x64 from https://github.com/actions/node-versions/releases/download/14.0.0-20200507.99/node-14.0.0-linux-x64.tar.gz
> Extracting ...
> /usr/bin/tar xz --strip 1 --warning=no-unknown-keyword -C /home/runner/work/_temp/de6b1539-a95c-46f0-9478-d114e0041d85 -f /home/runner/work/_temp/a219392d-3962-4c6f-9873-9a0f313a7c49
> Adding to the cache ...
> 
> Environment details
>   node: v14.0.0
>   npm: 6.14.4
>   yarn: 1.22.19
> ```
> 
> Link to the error: https://github.com/elastic/apm-agent-nodejs/actions/runs/5575905829/jobs/10187134325

Mystery solved, `npm` leaked into this PR's `package.json` although I do not remember adding it as a dependency

**Comment by trentm** [comment]
Sorry for your merge pain. Might be easiest to do a merge like this:
- copy in the state of the file on this PR before re-merging with main
- run `npm run lint:fix` to update its styling
- `git add ...` that file

**Comment by david-luna** [comment]
Closing this PR to prepare this instrumentation for next major release

**Pull request closed without merging**